src/api.js

```diff
@@ -211,11 +211,26 @@ var api = {
       ? Promise.resolve({ access_token: accessToken })
       : require("./auth").getAccessToken(refreshToken, commandScopes);
   },
-  addRequestHeaders: function(reqOptions) {
-    // Runtime fetch of Auth singleton to prevent circular module dependencies
+  addRequestHeaders: function(reqOptions, options) {
     _.set(reqOptions, ["headers", "User-Agent"], "FirebaseCLI/" + CLI_VERSION);
     _.set(reqOptions, ["headers", "X-Client-Version"], "FirebaseCLI/" + CLI_VERSION);
-    return api.getAccessToken().then(function(result) {
+
+    var secureRequest = true;
+    if (options && options.origin) {
+      // Only 'https' requests are secure. Protocol includes the final ':'
+      // https://developer.mozilla.org/en-US/docs/Web/API/URL/protocol
+      const originUrl = url.parse(options.origin);
+      secureRequest = originUrl.protocol === "https:";
+    }
+
+    // For insecure requests we send a special 'owner" token which the emulators
+    // will accept and other secure APIs will deny.
+    var getTokenPromise = secureRequest
+      ? api.getAccessToken()
+      : Promise.resolve({ access_token: "owner" });
+
+    // Runtime fetch of Auth singleton to prevent circular module dependencies
```

Comment:
What is this comment for?

Reply:
I don't know it was there before and I didn't understand it well enough to remove it.  I think it's because `getAccessToken()` contains `require(./auth)`

Reply:
I'll move it.